README.md

```diff
@@ -225,6 +225,35 @@ await fastify.register(require('@fastify/swagger-ui'), {
 })
 ```
 
+#### Protect your documentation routes
+
+You can protect your documentation by configuring an authentication hook.
+Here is an example using the [`@fastify/basic-auth`](https://github.com/fastify/fastify-basic-auth) plugin:
+
+##### Example
+```js
+const fastify = require('fastify')()
+
+fastify.register(require('@fastify/swagger'))
+
+await fastify.register(require('@fastify/basic-auth'), {
+  validate (username, password, req, reply, done) {
+    if (username === 'admin' && password === 'admin') {
```

Comment:
This is actually prone to timing attacks. We should either write above a comment that this is just for demonstration purposes and should be implemented in a safe manner OR directly write it properly.

See also
https://github.com/fastify/fastify-basic-auth/blob/master/example/example.js

Maybe we should also improve the code snippets in the readme of fastify basic auth

Reply:
This should be used https://nodejs.org/dist/latest-v18.x/docs/api/crypto.html#cryptotimingsafeequala-b

Reply:
Yes, we do not want to write examples that are prone to security issues when copied and pasted. Way too many people copy and paste without reading the caveats. Specifically, the example should follow https://github.com/fastify/fastify-basic-auth/blob/7db0266f096e15029ec118975b27335d412e3235/example/example.js#L13-L23

Reply:
Thank you for your suggestions, I've updated the docs to reflect this.